code_generation/templates/src/power_grid_model/metadata_enums.py.jinja

```diff
@@ -0,0 +1,77 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""Data types for power grid model dataset and component types."""
+
+# This file is automatically generated. DO NOT modify it manually!
+
+import sys
+from typing import Literal, Union
+
+# To avoid conflicts with src/power_grid_model/enum.py
+# pylint: disable=wrong-import-position
+
+sys_path = sys.path.pop(0)
+from enum import Enum
+
+# pylint: enable=wrong-import-position
+
+sys.path.append(sys_path)
+
+# Value names are defined in lower case instead of upper case
+# pylint: disable=invalid-name
+
+
+class DataTypes(Enum):
+    """Dataset types."""
+
+    {% for dataset_type in dataset_types %}
+    {{ dataset_type }} = "{{ dataset_type }}"
+    {% endfor %}
+
+
+DataTypesLiteral = Literal[
+    {% for dataset_type in dataset_types %}
+    "{{ dataset_type }}",
+    {% endfor %}
+]
+
+
+DataType = Union[DataTypes, DataTypesLiteral]
+"""
+A DataType is the type of a :class:`BatchDataset`.
+
+- Examples: 
+
+    - DataType.input = "input"
+    - DataType.update = "update"
```

Comment:
Awaiting discussion with the rest of the team, but in principle I think we should only place examples of the new datatypes as it may lead to confusion because mixing both cannot be done if we want any sort of type annotations.

Reply:
alright 👍